src/gui_common/charts/line/DataPoint.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using Godot;
+
+/// <summary>
+///   Point / marker on a line chart containing a single data value.
+/// </summary>
+public class DataPoint : Control
+{
+    private Texture graphMarkerCircle;
+    private Texture graphMarkerCross;
+
+    private bool isMouseOver;
+
+    private Vector2 coordinate;
+    private float size;
+
+    public DataPoint(float xValue, float yValue, float size = 8f, MarkerIcon shape = MarkerIcon.Circle,
+        bool drawAtZeroValue = false)
+    {
+        Value = new Vector2(xValue, yValue);
+        Size = size;
+        IconType = shape;
+        DrawAtZeroValue = drawAtZeroValue;
+    }
+
+    public enum MarkerIcon
+    {
+        Circle,
+        Cross,
+    }
+
+    /// <summary>
+    ///   Visual shape of the point
+    /// </summary>
+    public MarkerIcon IconType { get; set; }
+
+    /// <summary>
+    ///   Actual data the point represents
+    /// </summary>
+    public Vector2 Value { get; set; }
+
+    /// <summary>
+    ///   Position of the point on the chart, this is different from Value
+    /// </summary>
+    public Vector2 Coordinate
+    {
+        get => coordinate;
+        set
+        {
+            coordinate = value;
+            RectPosition = value - (RectSize / 2);
+        }
+    }
+
+    /// <summary>
+    ///   Visual size of the point
+    /// </summary>
+    public float Size
+    {
+        get => size;
+        set
+        {
+            size = value;
+
+            // Increased by 10 for a more bigger cursor detection area
+            RectSize = new Vector2(value + 10, value + 10);
+        }
+    }
+
+    public Color MarkerColor { get; set; }
+
+    public bool Draw { get; set; }
+
+    /// <summary>
+    ///   The Draw property is set to false if the point has zero value in the LineChart by default,
+    ///   so this is used to flag the marker visual to keep being drawn.
+    /// </summary>
+    public bool DrawAtZeroValue { get; set; }
+
+    public override void _Ready()
+    {
+        graphMarkerCircle = GD.Load<Texture>("res://assets/textures/gui/bevel/graphMarkerCircle.png");
+        graphMarkerCross = GD.Load<Texture>("res://assets/textures/gui/bevel/graphMarkerCross.png");
+
+        Connect("mouse_entered", this, nameof(OnMouseEnter));
+        Connect("mouse_exited", this, nameof(OnMouseExit));
+
+        Update();
+    }
+
+    public override void _Draw()
+    {
+        if (!Draw)
+            return;
```

Comment:
Would it work to just set Visibility to false on all the various things, instead of setting Draw to true / false?
I'm wondering why you chose this design?

Reply:
Mainly because I want the tooltips to still be hoverable for the data points at zero coordinates, since setting Visibility to false also disable the control's mouse enter/exit detection. But yeah I agree that this is unnecessarily messy and Visibility property should be used instead.

Reply:
Okay, so this is needed because of tooltips. If this design needs to be kept, you could add a comment to the Draw property to say why it is needed.